plugins/modules/update_systems.py

```diff
@@ -0,0 +1,219 @@
+"""
+
+EXAMPLES = """
+- name: Update systems
+  hosts: CRC
+  gather_facts: false
+  tasks:
+    - name: Update systems to latest version using ids
+      consoledot.edgemanagement.update_systems:
+        uuids:
+          - '160adfc2-00b1-48d5-b607-8b5ae7f4a861'
+        version: 7
+      register: output
+
+- name: Filter systems
+  hosts: CRC
+  gather_facts: false
+  tasks:
+    - name: Filter systems by insight facts
+      consoledot.edgemanagement.filter_systems:
+        host_type: 'edge'
+        ipv4: '192.168.122.[22:199]'
+        display_name: 'frontendv8'
+        os_release: '8.5'
+        cores_per_socket: 1
+        infrastructure_vendor: 'kvm'
+        number_of_cpus: 1
+        number_of_sockets: 1
+        os_kernel_version: '4.18.0'
+        enabled_services:
+          - 'firewalld'
+          - 'rhcd'
+        installed_services:
+          - 'rhsm'
+          - 'rhcd'
+      register: filtered_systems
+
+    - name: Update systems to latest version using ids
+      consoledot.edgemanagement.update_systems:
+        systems: '{{ filtered_systems["matched_systems"] }}'
+        version: 7
+      register: output
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_text
+
+from ansible_collections.consoledot.edgemanagement.plugins.module_utils.edgemanagement import (
+    ConsoleDotRequest,
+    EDGE_API_UPDATES,
+    EDGE_API_IMAGESETS
+)
+import json
+
+
+def batch_edge_systems(systems: list, isUUIDS=True) -> dict:
+    if isUUIDS:
+        image_set_key = 'ImageSetID'
+        system_id_key = 'DeviceUUID'
+    else:
+        image_set_key = 'edge_image_set_id'
+        system_id_key = 'id'
+
+    image_set_ids = {}
+    for system in systems:
+        if system[image_set_key] in image_set_ids:
+            image_set_ids[system[image_set_key]].append(system[system_id_key])
+        else:
+            image_set_ids[system[image_set_key]] = [system[system_id_key]]
+    return image_set_ids
+
+
+def main():
+
+    argspec = dict(
+        systems=dict(required=False, type="list", elements="dict"),
+        uuids=dict(required=False, type="list", elements="str"),
+        groups=dict(required=False, type="list", elements="str"),
+        version=dict(required=False, type="int"),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argspec,
+    )
+
+    crc_request = ConsoleDotRequest(module)
+
+    def update_systems(systems, commit_id=0):
+        for image_set_id, system_uuids in systems.items():
+            system_post_data = {
+                'CommitID': commit_id,
+                'DevicesUUID': system_uuids
+            }
+            try:
+                crc_request.post(
+                    EDGE_API_UPDATES, data=json.dumps(system_post_data)
+                )
+            except Exception as e:
+                module.fail_json(msg=e)
+
+    dispatched_updated = False
+
+    # used with filter systems module
+    if module.params['systems']:
```

Comment:
I would suggest some `try`/`except` magic in here to validate that the data passed to the module is in fact the format expected (i.e. - did actually come from the `filter_systems` module), otherwise fail the task gracefully with some useful output about malformed data inputs passed to the module.